hexgame-rs/src/game/board.rs

```diff
@@ -0,0 +1,300 @@
+use super::cell::Cell;
+use super::cell::Ownership;
+
+/*
+ A board is a set of exagonal cells stacked in a 2D matrix shape.
+ It can be indexed similarly to a matrix with (x, y) coordinates.
+ The (0,0) coordinate starts at the bottom left corner of the board.
+ The x coordinate indexes horizontally (going from left to right)
+ and the y vertically (going from bottom to top)
+ The exagonal cells make so that the board is slanted in a
+ romboidal shape. The slanting goes to the left.
+ ----------
+ '         '
+  '         '
+   '         '
+     ----------
+*/
+
+#[derive(Debug)]
+pub struct Board {
+    pub dim_x: usize,
+    pub dim_y: usize,
+    pub cells: Vec<Vec<Cell>>,
+}
+
+impl Board {
+    pub fn new() -> Self {
+        Self {
+            dim_x: 0,
+            dim_y: 0,
+            cells: vec![vec![]],
+        }
+    }
+
+    pub fn new_from_dims(dim_x: usize, dim_y: usize) -> Self {
+        let mut matrix = Vec::with_capacity(dim_x);
```

Comment:
Mmm I mean I do not know rust and would need to search better. But what I think I am find confusing is that there are two concept here: fixed lengtjh and constant. So vectors have dynamic length (you can push to them) and also allow for complex type that change state.
The question is, is there a data structure that allows mutable and also fixed length?
From here: https://stackoverflow.com/questions/31360993/what-is-the-proper-way-to-initialize-a-fixed-length-array

Looks like it might be possible, but maybe no.

Reply:
Mhh, maybe I stopped to soon the research.
Looking at [this general article](https://dev.to/anilkhandei/mutable-arrays-in-rust-1k5o) and thinking about it some more, it should be possible.

Reply:
Trying a bit I found that since `cells` is an attribute of a class when we declare the class attributes if it is an array it has to be initialized immediately, but with a vector, you can wait and initialize at the time of construction since you don't really know the dimensions of the board that is going to be created beforehand.
Maybe there is an alternative way, but for now let''s stick with vectors and data on the heap rather than stack.